sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -566,6 +581,30 @@ public static bool GetIsAllDataRequested(this Activity activity)
             return GetAllDataRequestedMethod(activity);
         }
 
+        public static void SetDisplayName(this Activity activity, string? displayName)
```

Comment:
nit: 
```suggestion
        public static void SetDisplayName(this Activity activity, string displayName)
```

Reply:
Agreed this should not be nullable.

Reply:
looks like `ActivityAdaptor ` does not allow "non-nullable" value if not set in the constructor.

Reply:
the parameter should not be nullable. The field can remain nullable.